src/libostree/ostree-repo-checkout.c

```diff
@@ -1006,6 +1022,19 @@ checkout_tree_at (OstreeRepo                        *self,
       g_assert (options->force_copy);
     }
 
+  /* Uncompressed archive caches; should be considered deprecated */
```

Comment:
Can you explain why we need this chunk here? We'll try to create & open it on demand in `checkout_object_for_uncompressed_cache` anyway, right? Or is this just to get non-`ENOENT` failures early on? (In that case, let's mention that in the comment?)

Reply:
We also need to handle the case of reusing an existing cachedir before we try to create objects there.